### ISSUE TITLE
[BUG] Following update to 1.8.0 pwnagochi.service fails to start

### ISSUE THREAD

**Issue by joejsullivan** [aluminum-ice/pwnagotchi#63]
**Describe the bug**
It appears following the 1.8.0 update, pwnagochi does not start. The log shows an update available, downloaded, then applied. I wouldn't rule out that I'm doing something wrong.

```
[INFO] [update] checking for updates ...
[2023-11-05 08:17:36,815] [WARNING] update for aluminum-ice/pwnagotchi available (local version is '1.7.9'): https://github.com/aluminum-ice/pwnagotchi/archive/1.8.0.zip
[2023-11-05 08:17:36,817] [INFO] [update] downloading https://github.com/aluminum-ice/pwnagotchi/archive/1.8.0.zip to /tmp/updates/pwnagotchi/pwnagotchi_1.8.0.zip ...
[2023-11-05 08:17:39,600] [INFO] [update] extracting /tmp/updates/pwnagotchi/pwnagotchi_1.8.0.zip to /tmp/updates/pwnagotchi ...
[2023-11-05 08:17:40,937] [INFO] [update] installing pwnagotchi ...
[2023-11-05 08:18:13,779] [INFO] [update] done
[2023-11-05 08:18:13,809] [WARNING] rebooting ...
[2023-11-05 08:18:23,850] [WARNING] syncing...
```

**To Reproduce**
1. Flash 1.7.9
2. Restart will force the udpate
3. Pwnagotchi will fail to start

**Expected behavior**
Service should start and run

**Screenshots**

**Environment (please complete the following information):**
 - Pwnagotchi version 1.8.0
 - Raspbian GNU/Linux 10 (buster)
 - Linux pwnagotchi 5.10.103-v7+ #1529 SMP Tue Mar 8 12:21:37 GMT 2022 armv7l GNU/Linux
 - pi 2 W
 - Waveshare 3
 - piSugar 2

**Additional context**
Output of trying to start service:

```
 pwnagotchi
Traceback (most recent call last):
  File "/usr/local/bin/pwnagotchi", line 152, in <module>
    plugins.load(config)
  File "/usr/local/lib/python3.7/dist-packages/pwnagotchi/plugins/__init__.py", line 124, in load
    enabled = [name for name, options in config['main']['plugins'].items() if
  File "/usr/local/lib/python3.7/dist-packages/pwnagotchi/plugins/__init__.py", line 125, in <listcomp>
    'enabled' in options and options['enabled']]
TypeError: argument of type 'int' is not iterable
```

**Comment by Slayingripper** [comment]
I am facing the exact same issue, it was working fine all day yesterday with no issues, but after the update was applied it stopped working. I've gone through my Config file , even trying to use the default but the problem still persists.

Going through the log file : 

```
[2023-11-05 01:37:42,823] [INFO] [update] extracting /tmp/updates/pwnagotchi/pwnagotchi_1.8.0.zip to /tmp/updates/pwnagotchi ...
[2023-11-05 01:37:44,971] [INFO] [update] installing pwnagotchi ...

[2023-11-05 01:38:31,752] [INFO] [update] done
```

That was the last time it worked , so its definitely something to do with the update.

**Comment by Slayingripper** [comment]
I've rolled back to the previous version for now and turned off auto update until further notice

**Comment by DW4y** [comment]
Have the same issue. I yesterday everything works.

I have set up a new pwnagotchi three times with min. and max. config. All works fine until it updates, after this the error: `ExecStart=/usr/bin/pwnagotchi-launcher (code=exited, status=1/FAILURE)`

Same Setup:

- pi 2 W
- Waveshare 3
- piSugar 2

**Comment by aluminum-ice** [comment]
Do you see mon0?

**Comment by joejsullivan** [comment]
Auto update is still on. I re flashed, restarted and got nabbed by the update again.


```
[2023-11-05 20:29:10,667] [INFO] [update] checking for updates ...
[2023-11-05 20:29:13,496] [WARNING] update for aluminum-ice/pwnagotchi available (local version is '1.7.9'): https://github.com/aluminum-ice/pwnagotchi/archive/1.8.0.zip
[2023-11-05 20:29:13,507] [INFO] [update] downloading https://github.com/aluminum-ice/pwnagotchi/archive/1.8.0.zip to /tmp/updates/pwnagotchi/pwnagotchi_1.8.0.zip ...
[2023-11-05 20:29:16,373] [INFO] [update] extracting /tmp/updates/pwnagotchi/pwnagotchi_1.8.0.zip to /tmp/updates/pwnagotchi ...
[2023-11-05 20:29:17,240] [INFO] [update] installing pwnagotchi ...
[2023-11-05 20:29:50,187] [INFO] [update] done
[2023-11-05 20:29:50,217] [WARNING] rebooting ...
[2023-11-05 20:30:00,257] [WARNING] syncing...
```

Result post restart:
```
Traceback (most recent call last):
  File "/usr/local/bin/pwnagotchi", line 152, in <module>
    plugins.load(config)
  File "/usr/local/lib/python3.7/dist-packages/pwnagotchi/plugins/__init__.py", line 124, in load
    enabled = [name for name, options in config['main']['plugins'].items() if
  File "/usr/local/lib/python3.7/dist-packages/pwnagotchi/plugins/__init__.py", line 125, in <listcomp>
    'enabled' in options and options['enabled']]
TypeError: argument of type 'int' is not iterable
```

**Comment by aluminum-ice** [comment]
That seems like an issue with one of the plugins. I am going to remove v1.8.0 for now.

**Comment by aluminum-ice** [comment]
@retiolus Could you take a look at this issue? It might be something with the new plugins I did not catch in my review.

**Comment by retiolus** [comment]
Hi, I'll have a look, but with the error log I don't get which plugin is making this error...

**Comment by retiolus** [comment]
And sorry if the problem comes from one of my plugins, adding them is not documented in the docs. Is it mandatory to add default values in config.toml? I haven't done that, maybe the problem comes from here?

**Comment by aluminum-ice** [comment]
@retiolus tbh I'm not sure it's the plugins but going through Joe's log report it seems like it's an issue with a plugin. And I only merged your plugin PRs (no other major changes). I didn't see anything obvious in the code. I'm wondering if you fork my current repo, compile and run it on your pwnagotchi, do you see an error? I didn't. 

If you do though, can you remove the new plugins one by one and see which is the cause? I'm tied up with other stuff and won't get to this any time soon.

**Comment by retiolus** [comment]
I just made a pull request to add default config for mastodon and ntfy plugins #66 and fixed a typo in the same config for a wpa-sec parameter I added [da12b10](https://github.com/aluminum-ice/pwnagotchi/pull/66/commits/da12b10c0a80e4d0d20ee3e4c094ae6ca497a0d6)

Maybe this will fix the error @joejsullivan

**Comment by aluminum-ice** [comment]
Resolved in v1.8.1
